pkg/northbound/proto/roles.proto

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019-present Open Networking Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package proto;
+
+// DeviceRoleConfig carries the template configuration associated with a device role
+message DeviceRoleConfig {
+    string role = 1;
```

Comment:
probably too early to make this an enumerated list

Reply:
Yes. Also, the idea is that operators may have notion or arbitrary roles, so this needs to stay open-ended.